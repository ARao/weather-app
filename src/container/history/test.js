import React from 'react';
import { shallow } from 'enzyme';
import ConnectedHistory from './index';
import configureStore from 'redux-mock-store'
// import { fetchHistories } from '../../actions/historyActions'



describe('History connectd component', () => {
   const fetchHistories = jest.fn()
  const initialState = {
    weather :{
      histories : [
        {
        "forecast": {
          "forecastday": [
            {
                "date": "2019-01-03",
                "date_epoch": 1546473600,
                "day": {
                    "maxtemp_c": 23.5,
                    "mintemp_c": 14.9,
                    "mintemp_f": 58.8,
                  }
                }
              ]
            }
          }  
        ]
      }
    }
  const mockStore  =  configureStore();
  let wrapper, store;
  beforeEach(()=>{
    store = mockStore(initialState);
    wrapper = shallow(<ConnectedHistory store={ store } fetchHistories={ fetchHistories } /> )
  })

  it('++++ render the connected(SMART) component', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('++++ store should match', () => {
    expect(wrapper.props().histories ).toEqual(initialState.weather.histories)
  });

  it('+++ snapshot match', () => {
      const renderedValue =  shallow(<ConnectedHistory store={store} fetchHistories={ fetchHistories } />)
      expect(renderedValue).toMatchSnapshot();
  });
});