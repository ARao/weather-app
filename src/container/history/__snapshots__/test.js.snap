// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History connectd component +++ snapshot match 1`] = `
<History
  fetchHistories={[Function]}
  histories={
    Array [
      Object {
        "forecast": Object {
          "forecastday": Array [
            Object {
              "date": "2019-01-03",
              "date_epoch": 1546473600,
              "day": Object {
                "maxtemp_c": 23.5,
                "mintemp_c": 14.9,
                "mintemp_f": 58.8,
              },
            },
          ],
        },
      },
    ]
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
